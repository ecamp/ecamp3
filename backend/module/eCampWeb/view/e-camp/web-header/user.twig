{% extends 'e-camp/web-header/base.twig' %}

{% block header_icon %}
    <table border="0" style="width: 100%; border: 0;">
        <tr>
            <td width="1" style="padding-right: 10px;">
                <i class="zmdi zmdi-hc-fw zmdi-hc-3x zmdi-account"></i>
            </td>
            <td style="padding-top: 3px; padding-bottom: 5px">
                <h1>{{ user.username }}</h1>
            </td>
        </tr>
    </table>
{% endblock %}

{% block breadcrumb %}
    {{ parent() }}

    <li class="breadcrumb-item">
        <a href="{{ url('ecamp.web/user', { 'userId': user.id }) }}">
            <i class="zmdi zmdi-account"></i> {{ user.username }}
        </a>
    </li>
{% endblock %}

{% block hr %}
    <hr style="margin-top: 5px; margin-bottom: 5px;" />
{% endblock %}