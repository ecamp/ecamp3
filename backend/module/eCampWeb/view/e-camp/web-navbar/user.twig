<section class="container">
    <div class="row" style="margin-bottom: 10px">
        <div class="col-sm-12">
            <div class="nav nav-justified btn-group btn-block" role="group">
                <a class="nav-item btn btn-sm btn-outline-secondary"
                   href="{{ url('ecamp.web/user', { user: user }) }}"
                >
                    <i class="zmdi zmdi-hc-fw zmdi-info-outline"></i> Info
                </a>
                <a class="nav-item btn btn-sm btn-outline-secondary d-none d-sm-block"
                   href="{{ url('ecamp.web/user/membership', { user: user }) }}"
                >
                    <i class="zmdi zmdi-hc-fw zmdi-accounts"></i> Groups
                </a>
                <a class="nav-item btn btn-sm btn-outline-secondary d-none d-sm-block"
                   href="{{ url('ecamp.web/user/camps', { user: user }) }}"
                >
                    <i class="zmdi zmdi-hc-fw zmdi-labels"></i> Camps
                </a>
                <a class="nav-item btn btn-sm btn-outline-secondary"
                   href="{{ url('ecamp.web/user/friends', { user: user }) }}"
                >
                    <i class="zmdi zmdi-hc-fw zmdi-accounts"></i> Friends
                </a>

                <a href="#"
                   class="btn btn-sm btn-outline-secondary dropdown-toggle"
                   style="width: 60px"
                   data-toggle="dropdown"
                   aria-haspopup="true"
                   aria-expanded="false"
                >
                </a>
                <div class="dropdown-menu dropdown-menu-right">
                    <a class="dropdown-item d-block d-sm-none"
                       href="{{ url('ecamp.web/user/membership', { user: user }) }}"
                    >
                        <i class="zmdi zmdi-hc-fw zmdi-accounts"></i> Groups
                    </a>
                    <a class="dropdown-item d-block d-sm-none"
                       href="{{ url('ecamp.web/user/camps', { user: user }) }}"
                    >
                        <i class="zmdi zmdi-hc-fw zmdi-labels"></i> Camps
                    </a>

                    <div class="dropdown-divider d-block d-sm-none"></div>

                    <a class="dropdown-item"
                       href="{# {{ url('ecamp.web/user/admin', { user: user }) }} #}"
                    >
                        <i class="zmdi zmdi-hc-fw zmdi-settings"></i> Admin
                    </a>
                </div>
                {#
                #}
            </div>
        </div>
    </div>
</section>