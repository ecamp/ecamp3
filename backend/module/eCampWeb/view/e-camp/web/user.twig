{% extends 'e-camp/web-layout/layout.base.twig' %}

{% block setTitle %}
    {{ parent() }}
    {% do headTitle().append(user.username) %}
{% endblock %}

{% block layout_header %}
    {% include 'e-camp/web-header/user.twig' %}
    {% include 'e-camp/web-navbar/user.twig' %}
{% endblock %}
