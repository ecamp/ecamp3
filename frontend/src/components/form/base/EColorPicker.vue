<!--
Displays a field as a color picker (can be used with v-model)
-->
<template>
  <base-picker
    icon="mdi-palette"
    :icon-color="value"
    :value="value"
    v-bind="$attrs"
    :vee-id="veeId"
    :parse-picker="parsePicker"
    :vee-rules="veeRules || { regex: /#([a-f0-9]{3}){1,2}\b/i }"
    :required="required"
    button-aria-label-i18n-key="components.form.base.eColorPicker.openPicker"
    open-on-text-field-click
    @input="$emit('input', $event)"
  >
    <template #prepend="{ color, attrs, on }">
      <ColorSwatch :color="color" class="mt-n1" v-bind="attrs" v-on="on" />
    </template>
    <template #default="picker">
      <v-card :style="{ '--picker-contrast-color': contrast }" data-testid="colorpicker">
        <v-color-picker
          :value="removeAlpha(picker.value || '')"
          flat
          @input="picker.onInput"
        />
        <v-divider />
        <div class="d-flex gap-2 pa-4 flex-wrap">
          <ColorSwatch
            v-for="swatch in swatches"
            :key="swatch"
            :color="swatch"
            @select-color="picker.onInput"
          />
        </div>
      </v-card>
    </template>

    <!-- passing the append slot through -->
    <template #append>
      <slot name="append" />
    </template>
  </base-picker>
</template>

<script>
import BasePicker from './BasePicker.vue'
import { formComponentMixin } from '@/mixins/formComponentMixin.js'
import { contrastColor } from '@/common/helpers/colors.js'
import ColorSwatch from '@/components/form/base/ColorPicker/ColorSwatch.vue'

export default {
  name: 'EColorPicker',
  components: { ColorSwatch, BasePicker },
  mixins: [formComponentMixin],
  props: {
    value: { type: String, required: true },
  },
  data: () => ({
    swatches: [
      '#90B7E4',
      '#6EDBE9',
      '#4dbb52',
      '#FF9800',
      '#FD7A7A',
      '#d584e9',
      '#BBBBBB',

      '#1964B1',
      '#1E86CA',
      '#3DB842',
      '#F1810D',
      '#C71A1A',
      '#CF3BD6',
      '#575757',
    ],
  }),
  computed: {
    contrast() {
      // Vuetify returns invalid value #NANNAN in the initialization phase
      return this.value && this.value !== '#NANNAN' ? contrastColor(this.value) : 'black'
    },
  },
  methods: {
    /**
     * Format internal value for display in the UI
     */
    format(val) {
      return val || ''
    },
    parsePicker(val) {
      const result = this.removeAlpha(val)
      if (result.toLowerCase() === this.value.toLowerCase()) {
        // Avoid changing the case if that is all that has changed
        return Promise.resolve(this.value)
      }
      return Promise.resolve(result)
    },
    removeAlpha(hex) {
      return hex.length === 9 ? hex.substring(0, 7) : hex
    },
    update(picker, value) {
      picker.on.input(value)
    },
  },
}
</script>

<style scoped>
:deep(.v-color-picker__dot > div::before) {
  content: 'â€¢';
  color: var(--picker-contrast-color);
  display: block;
  width: 100%;
  line-height: 26px;
  font-size: 28px;
  text-align: center;
}
</style>
