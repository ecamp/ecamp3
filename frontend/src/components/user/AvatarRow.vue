<template>
  <div class="avatarrow" :style="avatarrow">
    <div v-for="user in users" :key="user.id" class="avataritem">
      <UserAvatar :size="Number(size)" :user="user" />
    </div>
  </div>
</template>

<script>
export default {
  name: 'AvatarRow',
  props: {
    users: { type: Array, default: () => [] },
    size: { type: [Number, String], default: 20 },
  },
  computed: {
    maxWidth() {
      return (this.users?.length - 1) * (Number(this.size) * 0.25) + Number(this.size)
    },
    avatarrow() {
      return {
        'max-width': `${this.maxWidth}px`,
        'font-size': `${this.size}px`,
      }
    },
  },
}
</script>

<style scoped lang="scss">
.avatarrow {
  display: flex;
  flex-direction: row-reverse;
  gap: 0.75em;
  padding-left: 0.5em;
  padding-right: 0.5em;
  transition: gap 0.25s ease;
}

@media #{map-get($display-breakpoints, 'md-and-up')} {
  .avatarrow {
    gap: 1.1em;
  }
}

.avatarrow:hover {
  gap: 1.1em;
}

.avataritem {
  display: grid;
  width: 0;
  place-content: center;
  text-decoration: none;
}
</style>
