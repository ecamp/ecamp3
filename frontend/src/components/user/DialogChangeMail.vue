<template>
  <dialog-form
    v-model="showDialog"
    :title="$tc('components.user.dialogChangeMail.title')"
    :submit-action="status === 'initial' ? sendChangeMailRequest : null"
    :cancel-action="close"
    cancel-label="global.button.close"
    :cancel-visible="status !== 'initial'"
    submit-color="success"
  >
    <template #activator="scope">
      <slot name="activator" v-bind="scope" />
    </template>
    <div v-if="status === 'initial'">
      <e-text-field
        v-model="entityData.newEmail"
        :name="$tc('entity.user.fields.email')"
        vee-rules="email|required"
        append-icon="mdi-at"
        autofocus
      />
      <p class="mt-5">
        {{ $tc('components.user.dialogChangeMail.message') }}
      </p>
    </div>
    <div v-if="status == 'success'">
      {{ $tc('components.user.dialogChangeMail.success') }}
    </div>
    <div v-if="status == 'error'">
      {{ $tc('components.user.dialogChangeMail.error') }}
    </div>
  </dialog-form>
</template>

<script>
import DialogForm from '@/components/dialog/DialogForm.vue'
import DialogBase from '@/components/dialog/DialogBase.vue'

export default {
  name: 'DialogChangeMail',
  components: { DialogForm },
  extends: DialogBase,
  data() {
    return {
      status: '',
      entityProperties: ['newEmail'],
    }
  },
  watch: {
    // copy data whenever dialog is opened
    showDialog: function (showDialog) {
      if (showDialog) {
        this.status = 'initial'
        this.loadEntityData(this.$store.state.auth.user.profile()._meta.self)
      }
    },
  },
  methods: {
    async sendChangeMailRequest() {
      await this.api
        .patch(this.entityUri, this.entityData)
        .then(() => {
          this.status = 'success'
        })
        .catch(() => {
          this.status = 'error'
        })
    },
  },
}
</script>
