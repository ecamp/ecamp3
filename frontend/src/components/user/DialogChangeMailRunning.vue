<template>
  <dialog-form
    v-model="showDialog"
    :title="$tc('components.user.dialogChangeMailRunning.title')"
    :cancel-action="status === 'initial' ? null : close"
    cancel-label="global.button.close"
  >
    <div v-if="status === 'initial'">
      <v-progress-circular indeterminate />
      {{ $tc('components.user.dialogChangeMailRunning.message') }}
    </div>
    <div v-if="status === 'success'">
      <v-icon>mdi mdi-check</v-icon>
      {{ $tc('components.user.dialogChangeMailRunning.success') }}
    </div>
    <div v-if="status === 'error'">
      {{ $tc('components.user.dialogChangeMailRunning.error') }}
    </div>
  </dialog-form>
</template>

<script>
import DialogForm from '@/components/dialog/DialogForm.vue'
import DialogBase from '@/components/dialog/DialogBase.vue'

export default {
  name: 'DialogChangeMailRunning',
  components: { DialogForm },
  extends: DialogBase,
  props: {
    emailVerificationKey: { type: String, required: false, default: null },
  },
  data() {
    return {
      status: '',
    }
  },
  watch: {
    showDialog: function (showDialog) {
      if (!showDialog) {
        this.$router.push({ name: 'profile' })
      }
    },
  },
  mounted() {
    if (this.emailVerificationKey) {
      this.status = 'initial'
      this.showDialog = true

      this.$auth
        .user()
        .profile()
        ._meta.load.then((p) => {
          p.$patch({
            untrustedEmailKey: this.emailVerificationKey,
          })
            .then(() => {
              this.status = 'success'
            })
            .catch(() => {
              this.status = 'error'
            })
        })
    }
  },
}
</script>
