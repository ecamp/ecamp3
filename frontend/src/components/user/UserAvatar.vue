<template>
  <v-avatar :size="size" :color="color" :title="displayName" v-bind="$attrs">
    <span :aria-hidden="shortAria ? null : 'true'" :style="style">{{ initials }}</span>
    <span v-if="shortAria" class="d-sr-only">{{ displayName }}</span>
  </v-avatar>
</template>
<script>
import campCollaborationInitials from '@/common/helpers/campCollaborationInitials.js'
import {
  contrastColor,
  userColor,
  campCollaborationColor,
  defaultColor,
} from '@/common/helpers/colors.js'
import { range } from '@/common/helpers/interpolation.js'
import userInitials from '@/common/helpers/userInitials.js'
import campCollaborationDisplayName from '@/common/helpers/campCollaborationDisplayName.js'
import userDisplayName from '@/common/helpers/userDisplayName.js'

export default {
  name: 'UserAvatar',
  props: {
    size: { type: [Number, String], required: false, default: 48 },
    user: { type: Object, default: null },
    campCollaboration: { type: Object, default: null },
    shortAria: { type: Boolean, default: false },
  },
  computed: {
    isLoading() {
      return (this.user || this.campCollaboration)?._meta?.loading
    },
    color() {
      if (this.isLoading) {
        return defaultColor()
      }
      return this.campCollaboration
        ? campCollaborationColor(this.campCollaboration)
        : userColor(this.user)
    },
    initials() {
      if (this.isLoading) {
        return ''
      }
      return this.campCollaboration
        ? campCollaborationInitials(this.campCollaboration)
        : userInitials(this.user)
    },
    displayName() {
      if (this.isLoading) {
        return ''
      }
      return this.campCollaboration
        ? campCollaborationDisplayName(this.campCollaboration, this.$tc.bind(this))
        : userDisplayName(this.user)
    },
    /**
     * Font size of small avatars should be half the size,
     * bigger avatars should have more visual padding
     */
    fontSize() {
      const ratio = range(16, 32, 2, 2.3, Number(this.size))
      return Number(this.size) / ratio
    },
    style() {
      return {
        color: contrastColor(this.color),
        fontSize: `${this.fontSize}px`,
        fontWeight: Number(this.size) > 44 ? 450 : Number(this.size) > 24 ? 500 : 600,
        letterSpacing: Number(this.size) < 24 ? '.05em' : 0,
        marginRight: Number(this.size) < 24 ? '-.125em' : 0,
      }
    },
  },
}
</script>
