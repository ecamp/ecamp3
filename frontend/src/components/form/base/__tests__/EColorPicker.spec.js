import { screen, waitFor } from '@testing-library/vue'
import { render, setTestLocale, snapshotOf } from '@/test/renderWithVuetify.js'
import user from '@testing-library/user-event'
import EColorPicker from '../EColorPicker.vue'

import { regex } from 'vee-validate/dist/rules'
import { extend } from 'vee-validate'
extend('regex', regex)

describe('An EColorPicker', () => {
  const COLOR1 = '#ff0000'
  const COLOR2 = '#ff00ff'
  const COLOR3 = '#FAFFAF'
  const INVALID_COLOR = 'some new color'
  const PICKER_BUTTON_LABEL_TEXT = 'Dialog öffnen um eine Farbe für test zu wählen'
  const VALIDATION_MESSAGE = 'test is not valid'

  beforeEach(() => {
    setTestLocale('de')
  })

  it('renders the component', async () => {
    // given

    // when
    render(EColorPicker, {
      props: {
        value: COLOR1,
        name: 'test',
      },
    })

    // then
    await screen.findByDisplayValue(COLOR1)
    screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
  })

  it('looks like a color picker', async () => {
    // given

    // when
    const { container } = render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })

    // then
    expect(snapshotOf(container)).toMatchSnapshot('pickerclosed')

    // when
    await user.click(screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT))

    // then
    await screen.findByText('Schliessen')
    expect(snapshotOf(container)).toMatchSnapshot('pickeropen')
  })

  it('opens the picker when the text field is clicked', async () => {
    // given
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)

    // when
    await user.click(inputField)

    // then
    await waitFor(async () => {
      expect(await screen.findByText('Schliessen')).toBeVisible()
    })
  })

  it('opens the picker when the icon button is clicked', async () => {
    // given
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)

    // when
    await user.click(button)

    // then
    await waitFor(async () => {
      expect(await screen.findByText('Schliessen')).toBeVisible()
    })
  })

  it('closes the picker when clicking the close button', async () => {
    // given
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByText('Schliessen')).toBeVisible()
    })
    const closeButton = screen.getByText('Schliessen')

    // when
    await user.click(closeButton)

    // then
    await waitFor(async () => {
      expect(await screen.queryByText('Schliessen')).not.toBeVisible()
    })
  })

  it('closes the picker when clicking outside', async () => {
    // given
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByText('Schliessen')).toBeVisible()
    })

    // when
    await user.click(document.body)

    // then
    await waitFor(async () => {
      expect(await screen.queryByText('Schliessen')).not.toBeVisible()
    })
  })

  it('closes the picker when pressing escape', async () => {
    // given
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByText('Schliessen')).toBeVisible()
    })

    // when
    await user.keyboard('{Escape}')

    // then
    await waitFor(async () => {
      expect(await screen.queryByText('Schliessen')).not.toBeVisible()
    })
  })

  it('does not close the picker when selecting a color', async () => {
    // given
    const { container } = render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByText('Schliessen')).toBeVisible()
    })

    // when
    // click inside the color picker canvas to select a different color
    const canvas = container.querySelector('canvas')
    await user.click(canvas, { clientX: 10, clientY: 10 })

    // then
    // close button should stay visible
    await expect(
      waitFor(() => {
        expect(screen.queryByText('Schliessen')).not.toBeVisible()
      })
    ).rejects.toThrow(/Received element is visible/)
  })

  it('updates v-model when the value changes', async () => {
    // given
    const { emitted } = render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)

    // when
    await user.clear(inputField)
    await user.keyboard(COLOR2)

    // then
    await waitFor(async () => {
      const events = emitted().input
      // some input events were fired
      expect(events.length).toBeGreaterThan(0)
      // the last one included the parsed version of our entered time
      expect(events[events.length - 1]).toEqual([COLOR2])
    })
    // Our entered time should be visible...
    screen.getByDisplayValue(COLOR2)
    // ...and stay visible
    await expect(
      waitFor(() => {
        expect(screen.getByDisplayValue(COLOR2)).not.toBeVisible()
      })
    ).rejects.toThrow(/Received element is visible/)
  })

  it('updates v-model when a new color is selected in the picker', async () => {
    // given
    const { emitted, container } = render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    await screen.findByDisplayValue(COLOR1)
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)

    // when
    // click the button to open the picker
    await user.click(button)
    // click inside the color picker canvas to select a different color
    const canvas = container.querySelector('canvas')
    await user.click(canvas, { clientX: 10, clientY: 10 })
    // click the close button
    await user.click(screen.getByText('Schliessen'))

    // then
    await waitFor(async () => {
      const events = emitted().input
      // some input events were fired
      expect(events.length).toBeGreaterThan(0)
      // the last one included the parsed version of our entered time
      expect(events[events.length - 1]).toEqual([COLOR3])
    })
    // Our entered time should be visible...
    screen.getByDisplayValue(COLOR3)
    // ...and stay visible
    await expect(
      waitFor(() => {
        expect(screen.getByDisplayValue(COLOR3)).not.toBeVisible()
      })
    ).rejects.toThrow(/Received element is visible/)
  })

  it('validates the input', async () => {
    // given
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)

    // when
    await user.clear(inputField)
    await user.keyboard(INVALID_COLOR)

    // then
    await screen.findByText(VALIDATION_MESSAGE)
  })

  it('accepts 3-digit hex color codes, after picker has been shown', async () => {
    render(EColorPicker, {
      props: { value: COLOR1, name: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    // click the button to open the picker
    await user.click(button)

    // when
    await user.clear(inputField)
    await user.keyboard('#abc')

    // then
    await waitFor(() => {
      screen.getByDisplayValue('#AABBCC')
    })
  })
})
