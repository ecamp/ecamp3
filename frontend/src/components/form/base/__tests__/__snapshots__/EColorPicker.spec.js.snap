// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`An EColorPicker > looks like a color picker > pickerclosed 1`] = `
<div data-app="true">
     <div class="e-form-container"
          name="test"
          vee-rules="[object Object]">
          <div class="v-menu">
               <div class="e-form-container">
                    <div class="v-input v-input--hide-details theme--light v-text-field v-text-field--filled v-text-field--enclosed">
                         <div class="v-input__prepend-outer">
                              <button aria-label="Dialog öffnen um eine Farbe für test zu wählen"
                                      class="v-icon notranslate v-icon--link mdi mdi-palette theme--light"
                                      style="color: rgb(255, 0, 0); caret-color: #ff0000;"
                                      type="button"></button>
                         </div>
                         <div class="v-input__control">
                              <div class="v-input__slot">
                                   <div class="v-text-field__slot">
                                        <label class="v-label theme--light"
                                               for
                                               style="left: 0px; position: absolute;">test</label>
                                        <input id
                                               type="text">
                                   </div>
                              </div>
                         </div>
                    </div>
               </div>
               <!---->
          </div>
     </div>
</div>
`;

exports[`An EColorPicker > looks like a color picker > pickeropen 1`] = `
<div data-app="true">
     <div class="e-form-container"
          name="test"
          vee-rules="[object Object]">
          <div class="e-form-container"
               value="#ff0000">
               <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--filled v-text-field--is-booted v-text-field--enclosed">
                    <div class="v-input__prepend-outer">
                         <button aria-label="Dialog öffnen um eine Farbe für test zu wählen"
                                 class="v-icon notranslate v-icon--link mdi mdi-palette theme--light"
                                 style="color: rgb(255, 0, 0); caret-color: #ff0000;"
                                 type="button"></button>
                    </div>
                    <div class="v-input__control">
                         <div class="v-input__slot">
                              <div class="v-text-field__slot">
                                   <label class="v-label v-label--active theme--light"
                                          for
                                          style="left: 0px; position: absolute;">test</label>
                                   <input id
                                          type="text">
                              </div>
                         </div>
                    </div>
               </div>
          </div>
          <div class="v-menu">
               <transition-stub name="scale-transition"></transition-stub>
          </div>
     </div>
     <div class="v-menu__content theme--light menuable__content__active "
          role="menu"
          style="max-height: auto; min-width: 290px; max-width: 290px; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
          <div id>
               <div class="v-card v-sheet theme--light">
                    <div class="v-color-picker v-sheet theme--light v-color-picker--flat theme--light"
                         style="max-width: 300px;">
                         <div class="v-color-picker__canvas"
                              style="width: 300px; height: 150px;">
                              <canvas height="150"
                                      width="300"></canvas>
                              <div class="v-color-picker__canvas-dot"
                                   style="width: 10px; height: 10px; transform: translate(295px, -5px);"></div>
                         </div>
                         <div class="v-color-picker__controls">
                              <div class="v-color-picker__preview v-color-picker__preview--hide-alpha">
                                   <div class="v-color-picker__dot">
                                        <div style="background: rgb(255, 0, 0);"></div>
                                   </div>
                                   <div class="v-color-picker__sliders">
                                        <div class="v-input v-color-picker__hue v-input--hide-details theme--light v-input__slider v-color-picker__track">
                                             <div class="v-input__control">
                                                  <div class="v-input__slot">
                                                       <div class="v-slider v-slider--horizontal theme--light">
                                                            <input disabled="disabled"
                                                                   id
                                                                   readonly="readonly"
                                                                   tabindex="-1"
                                                                   value="0">
                                                            <div class="v-slider__track-container">
                                                                 <div class="v-slider__track-background primary lighten-3"
                                                                      style="right: 0px; width: calc(100%);"></div>
                                                                 <div class="v-slider__track-fill primary"
                                                                      style="left: 0px; width: 0%;"></div>
                                                            </div>
                                                            <div aria-orientation="horizontal"
                                                                 aria-readonly="false"
                                                                 aria-valuemax="360"
                                                                 aria-valuemin="0"
                                                                 aria-valuenow="0"
                                                                 class="v-slider__thumb-container grey--text text--lighten-2"
                                                                 role="slider"
                                                                 style="left: 0%;"
                                                                 tabindex="0">
                                                                 <div class="v-slider__thumb grey lighten-2"></div>
                                                            </div>
                                                       </div>
                                                  </div>
                                             </div>
                                        </div>
                                   </div>
                              </div>
                              <div class="v-color-picker__edit">
                                   <div class="v-color-picker__input">
                                        <input max="255"
                                               min="0"
                                               step="1"
                                               type="number">
                                        <span>R</span>
                                   </div>
                                   <div class="v-color-picker__input">
                                        <input max="255"
                                               min="0"
                                               step="1"
                                               type="number">
                                        <span>G</span>
                                   </div>
                                   <div class="v-color-picker__input">
                                        <input max="255"
                                               min="0"
                                               step="1"
                                               type="number">
                                        <span>B</span>
                                   </div>
                                   <button class="v-btn v-btn--icon v-btn--round theme--light v-size--small"
                                           type="button">
                                        <span class="v-btn__content">
                                             <i aria-hidden="true"
                                                class="v-icon notranslate mdi mdi-unfold-more-horizontal theme--light"></i>
                                        </span>
                                   </button>
                              </div>
                         </div>
                    </div>
                    <button class="v-btn v-btn--text theme--light v-size--default primary--text"
                            type="button">
                         <span class="v-btn__content"> Schliessen </span>
                    </button>
               </div>
          </div>
     </div>
</div>
`;
